src/memory/db_stores/redis_store.rs

```diff
@@ -0,0 +1,233 @@
+        })
+    }
+
+    pub async fn clear_indexes(&self) -> Result<(), redis::RedisError> {
+        redis::cmd("FLUSHDB")
+            .query_async(&mut self.manager.clone())
+            .await
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct RedisStoreError(String);
+
+impl std::error::Error for RedisStoreError {}
+
+impl From<redis::RedisError> for RedisStoreError {
+    fn from(err: redis::RedisError) -> Self {
+        Self(err.to_string())
+    }
+}
+
+impl Display for RedisStoreError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Redis Memory Error: {}", self.0)
+    }
+}
+
+impl<
+    Address: Send + Sync + Hash + Eq + Debug + Clone + Deref<Target = [u8; ADDRESS_LENGTH]>,
+    const ADDRESS_LENGTH: usize,
+    const WORD_LENGTH: usize,
+> MemoryADT for RedisStore<Address, WORD_LENGTH>
+{
+    type Address = Address;
+    type Error = RedisStoreError;
+    type Word = [u8; WORD_LENGTH];
+
+    async fn batch_read(
+        &self,
+        addresses: Vec<Address>,
+    ) -> Result<Vec<Option<Self::Word>>, Self::Error> {
+        let refs: Vec<&[u8; ADDRESS_LENGTH]> =
+            addresses.iter().map(|address| address.deref()).collect();
+        self.manager
+            .clone()
+            .mget::<_, Vec<_>>(&refs)
+            .await
+            .map_err(Self::Error::from)
+    }
+
+    async fn guarded_write(
+        &self,
+        guard: (Self::Address, Option<Self::Word>),
+        bindings: Vec<(Self::Address, Self::Word)>,
+    ) -> Result<Option<Self::Word>, Self::Error> {
+        let (guard_address, guard_value) = guard;
+        let mut cmd = redis::cmd("EVALSHA")
+            .arg(self.write_script_hash.clone())
+            .arg(0)
+            .arg(&*guard_address)
+            .clone(); // Why cloning is necessary : https://stackoverflow.com/questions/64728534/how-to-resolve-creates-a-temporary-variable-which-is-freed-while-still-in-use
+        cmd = if let Some(byte_array) = guard_value {
+            cmd.arg(&byte_array).arg(bindings.len()).clone()
+        } else {
+            cmd.arg("false".to_string()).arg(bindings.len()).clone()
+        };
+        for (address, word) in bindings {
+            cmd = cmd.arg(&*address).arg(&word).clone();
+        }
+        cmd.query_async(&mut self.manager.clone())
+            .await
+            .map_err(|e| e.into())
+    }
+}
+
+#[cfg(feature = "test-utils")]
+#[cfg(test)]
+mod tests {
+
+    use serial_test::serial;
+
+    use super::*;
+    use crate::{
+        Address,
+        test::memory::{
+            test_guarded_write_concurrent, test_single_write_and_read, test_wrong_guard,
+        },
+    };
+
+    pub fn get_redis_url() -> String {
+        if let Ok(var_env) = std::env::var("REDIS_HOST") {
+            format!("redis://{var_env}:6379")
+        } else {
+            "redis://localhost:6379".to_string()
+        }
+    }
+
+    const TEST_ADR_WORD_LENGTH: usize = 16;
```

Comment:
Please no spelling in names, even for shortening them. `TEST_` brings no information, you can remove it instead.